src/repo.ts

```diff
@@ -1,56 +1,52 @@
 import fs from 'node:fs'
-import { chdir } from 'node:process'
+import path from 'node:path'
 import { Badge } from './badges.js'
-import { exec, execWithOutput } from './utils.js'
-
-export function syncRepo(owner: string, repo: string, token: string) {
-  if (fs.existsSync('repo')) {
-    chdir('repo')
-    exec('git', ['pull'])
-    chdir('..')
+import { $ as _, type TShellSync } from './utils.js'
+
+const MY_BADGES_NAME = 'My Badges'
+const MY_BADGES_EMAIL = 'my-badges@users.noreply.github.com'
+const MY_BADGES_FILE = 'my-badges/my-badges.json'
+const CWD = process.cwd()
+const DIR = 'repo'
+
+export const __internal__ = { cwd: CWD }
+const $ = ((...args: any) =>
+  _({
+    cwd: path.resolve(__internal__.cwd, DIR),
+    sync: true,
+  })(...args)) as TShellSync
+
+export function syncRepo(owner: string, repo: string, token?: string) {
+  const { cwd } = __internal__
+  if (fs.existsSync(path.resolve(cwd, DIR))) {
+    $`git pull`
     return
   }
+  const auth = token ? `${owner}:${token}@` : ''
 
-  exec('git', [
-    'clone',
-    '--depth=1',
-    `https://${owner}:${token}@github.com/${owner}/${repo}.git`,
-    'repo',
-  ])
-
-  chdir('repo')
-
-  exec('git', ['config', 'user.name', 'My Badges'])
-  exec('git', ['config', 'user.email', 'my-badges@users.noreply.github.com'])
-
-  chdir('..')
+  $({
+    cwd,
```

Comment:
To delegate git to create `repo` dir instead of:
```
fs.mkdirSync(path.resolve(cwd, DIR))
$`git clone --depth=1 https://${auth}github.com/${owner}/${repo}.git .`
```

Reply:
Next commands should also be with cmd? Also the default one also the same?

Reply:
No, `$` is preconfigured to interact with `cwd()/repo`
```ts
const $ = ((...args: any) =>
  _({
    cwd: path.resolve(__internal__.cwd, DIR),
    sync: true,
  })(...args)) as TShellSync
```